### PR TITLE
Kafka: Support for AWS IAM role auth

### PR DESCRIPTION
**Pull request by eunicecompra**
# Description

Follow up of the https://github.com/dapr/components-contrib/pull/3240

This is to add an AWS IAM role authentication aside from AWS credentials.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/components-contrib/issues/3239 (Already closed but related issue)

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by ItalyPaleAle — DISMISSED** [review]
LGTM @yaron2  can you please review too

**Comment by ItalyPaleAle** [comment]
@eunicecompra can you please fix the DCO?
https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work

**Comment by eunicecompra** [comment]
> @eunicecompra can you please fix the DCO? https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work

@ItalyPaleAle Missed this one. Done signing off my commits.

File: common/component/kafka/metadata.go [hunk 1]
```diff
@@ -237,6 +239,9 @@ func (k *Kafka) getKafkaMetadata(meta map[string]string) (*KafkaMetadata, error)
 		if m.AWSRegion == "" {
 			return nil, errors.New("missing AWS region property 'awsRegion' for authType 'awsiam'")
 		}
+		if m.AWSIamRoleArn == "" && m.AWSSecretKey == "" && m.AWSAccessKey == "" {
```
[the remark this answers is not in the document]
Comment (yaron2) [reply common/component/kafka/metadata.go:242]: Ack

**Review by yaron2 — APPROVED** [review]

File: common/component/kafka/metadata.go [hunk 2]
```diff
@@ -244,6 +246,9 @@ func (k *Kafka) getKafkaMetadata(meta map[string]string) (*KafkaMetadata, error)
...
```
Comment (famarting) [inline common/component/kafka/metadata.go:249]: Access key and secret key can actually be empty and its a supported feature, so the aws sdk picks the creds automatically from  other sources such as the environment. This new if condition breaks that feature

Comment (yaron2) [reply common/component/kafka/metadata.go:249]: cc @eunicecompra

Comment (eunicecompra) [reply common/component/kafka/metadata.go:249]: Yep! gotcha. I will remove this check.

**Pull request merged**